typing/env.ml

```diff
@@ -270,7 +282,8 @@ module TycompTbl =
       let keys2 = local_keys tbl2 [] in
       List.filter
         (fun id ->
-           is_local (find_same id tbl2) &&
+           let descr, _ = find_same id tbl2 in
+           is_local descr &&
```

Comment:
That feels wrong: it "works", but it's weird.
Here is the type of the function after this change:
```ocaml
val diff_keys: ('a -> bool) -> 'b t -> ('a * 'c) t -> Ident.t list
```
Before that it was:
```ocaml
val diff_keys: ('a -> bool) -> 'b t -> 'a t -> Ident.t list
```
which seems more natural.

I think it's the caller of this function that should be changed, while this function should remain untouched. (And the function appears to be called only once, so it shouldn't be too tedious to update)